src/map/Map.js

```diff
@@ -511,15 +518,20 @@ export var Map = Evented.extend({
 		return this;
 	},
 
-	// @method panInside(latlng: LatLng, options?: options): this
+	// @method panInside(latlng: LatLng, options?: padding options): this
 	// Pans the map the minimum amount to make the `latlng` visible. Use
-	// `padding`, `paddingTopLeft` and `paddingTopRight` options to fit
-	// the display to more restricted bounds, like [`fitBounds`](#map-fitbounds).
+	// padding options to fit the display to more restricted bounds.
 	// If `latlng` is already within the (optionally padded) display bounds,
 	// the map will not be panned.
 	panInside: function (latlng, options) {
 		options = options || {};
 
+		if ((options.padding && options.padding.length !== 2) ||

```

Comment:
Another occurrence of same code. Not good...

Reply:
Is the issue that I haven't factored out the error check into a separate fn? I can certainly do that if we decide these checks are enough for now.

Reply:
> if we decide these checks are enough for now.

I tend to think that current checks definitely not enough.
Why should we care padding options more that others?
And even if we do, why we do not care about similar occurrences:
- https://leafletjs.com/reference-1.7.1.html#marker-autopanpadding
- https://leafletjs.com/reference-1.7.1.html#popup-autopanpaddingtopleft (there are 3)
- etc

So I propose following:
1. Exclude validation code (and corresponding testes) from this PR, leave docs enhancement only (which alone solves most of original #7372 issue).
2. Make another PR adding point object validation code to `toPoint`.

Reply:
The other padding parameters you mention were explicitly documented as taking `Point`s. The `panInside` documentation originally lacked this explicit mention, which contributed to the original misunderstanding by the OP and the #7397 issue being raised. I suspect adding `toPoint` validation may end up catching fewer genuine issues while potentially adding a performance penalty for larger projects.

However, we can open that up for discussion on a separate PR! I'm happy to remove the options checks here and submit just the documentation updates for now.